### PR TITLE
feat: add request parameter to journey context

### PR DESCRIPTION
**Pull request by andrewvc**
This PR adds support for [API Requests](https://playwright.dev/docs/test-api-testing#sending-api-requests-from-ui-tests) in browser tests.

It makes using a plain HTTP client as simple as:

```ts
step('check body', async () => {
  const resp = await apiContext.get(params.url);
  expect((await resp.body()).toString()).toMatch(/Synthetics/);
})
```

Note, this is not a facility for writing pure API tests, but rather a way to support
writing plain HTTP requests in service of a browser based test (as indicated in the above-linked playwright docs).

As an example, a user may want to retrieve a set of credentials from a vault based on HTTP,
or get a piece of metadata from a JSON API.

While it would be possible for users to import `requests` from `playwright-core`
and achieve a similar effect this PR is superior in that:

1. It provides an officially sanctioned way to accomplish this common pattern.
2. It automatically initializes the client for users, saving a line of boilerplate
3. Users do not need to understand JS `import` syntax to use this common feature
4. It is available in `inline` journeys.

### CODE REVIEW

**Review by justinkambic — APPROVED** [review]
LGTM! Had one comment WRT naming but I'm also fine with merging as-is.

File: src/common_types.ts [hunk 1]
```diff
@@ -62,6 +63,7 @@ export type Driver = {
   context: ChromiumBrowserContext;
   page: Page;
   client: CDPSession;
+  apiContext: APIRequestContext;
```
[the remark this answers is not in the document]
Comment (andrewvc) [reply src/common_types.ts:66]: That said, for their proper API tests they call that object [`request`](https://playwright.dev/docs/test-api-testing#writing-tests)

WDYT about changing it to either `request` or `apiRequest`? Any pref? I'd prefer to call it `request` IMHO

[the remark this answers is not in the document]
Comment (dominiqueclarke) [reply src/common_types.ts:66]: `apiClient`?

**Comment by vigneshshanmugam** [comment]
@andrewvc Added the changes to your origin to keep the diff cleaner - https://github.com/andrewvc/synthetics-1/pull/1

File: __tests__/plugins/network.test.ts [hunk 2]
```diff
@@ -278,4 +278,18 @@ describe('network', () => {
       expect(timing.total).toBeLessThan(50);
     });
   });
+
+  it("doesn't capture network info from request context", async () => {
```
Comment (vigneshshanmugam) [inline __tests__/plugins/network.test.ts:282]: I checked PW and its not exposed correctly for our use case. Have to do a PR upstream to make it work.

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

**Pull request merged**